core/templates/services/platform-feature.service.spec.ts

```diff
@@ -0,0 +1,351 @@
+          'SAVED_FEATURE_FLAGS'))
+      ).toEqual({
+        timestamp: timestamp,
+        sessionId: sessionId,
+        featureStatusSummary: {
+          [FeatureNames.DummyFeature]: true,
+        }
+      });
+      expect(platformFeatureService.initialzedWithError).toBeFalse();
+    }));
+
+    it(
+      'should use SACSID instead of dev_appserver_login as sessionId when' +
+      ' saving results.', fakeAsync(() => {
+        const sessionId = 'session_id';
+        mockCookie(`SACSID=${sessionId}; dev_appserver_login=should_not_use`);
+
+        platformFeatureService = TestBed.get(PlatformFeatureService);
+
+        flushMicrotasks();
+
+        expect(
+          JSON.parse(windowRef.nativeWindow.sessionStorage.getItem(
+            'SAVED_FEATURE_FLAGS'))
+            .sessionId
+        ).toEqual(sessionId);
+      })
+    );
+
+    it(
+      'should use dev_app_server_login as sessionId when no SACSID is set',
+      fakeAsync(() => {
+        const sessionId = 'session_id';
+        mockCookie(`dev_appserver_login=${sessionId}`);
+
+        platformFeatureService = TestBed.get(PlatformFeatureService);
+
+        flushMicrotasks();
+
+        expect(
+          JSON.parse(windowRef.nativeWindow.sessionStorage.getItem(
+            'SAVED_FEATURE_FLAGS'))
+            .sessionId
+        ).toEqual(sessionId);
+      })
+    );
+
+    it('should load from sessionStorage if there\'s valid results.', fakeAsync(
+      () => {
+        const sessionId = 'session_id';
+        mockCookie(`SACSID=${sessionId}`);
+        mockSessionStore({
+          SAVED_FEATURE_FLAGS: JSON.stringify({
+            sessionId: sessionId,
+            timestamp: Date.now(),
+            featureStatusSummary: {
+              [FeatureNames.DummyFeature]: true,
+            }
+          })
+        });
+
+        tick(60 * 1000);
+        platformFeatureService = TestBed.get(PlatformFeatureService);
+
+        flushMicrotasks();
+
+        expect(apiService.fetchFeatureFlags).not.toHaveBeenCalled();
+        expect(platformFeatureService.initialzedWithError).toBeFalse();
+      })
+    );
+
+    it('should load from server if saved results have expired.',
+      fakeAsync(() => {
+        const sessionId = 'session_id';
+        mockCookie(`SACSID=${sessionId}`);
+        mockSessionStore({
+          SAVED_FEATURE_FLAGS: JSON.stringify({
+            sessionId: sessionId,
+            timestamp: Date.now(),
+            featureStatusSummary: {
+              [FeatureNames.DummyFeature]: true,
+            }
+          })
+        });
+
+        tick(13 * 3600 * 1000); // 13 hours later.
+        platformFeatureService = TestBed.get(PlatformFeatureService);
+
+        flushMicrotasks();
+
+        expect(apiService.fetchFeatureFlags).toHaveBeenCalled();
+        expect(platformFeatureService.initialzedWithError).toBeFalse();
+      })
+    );
+
+    it(
+      'should load from server if sessionId of saved result does not match.',
```

Comment:
Do you also need a test to verify that the server is not called if the session does match?

Reply:
Reminder to consider this comment.